regulation-worker/internal/delete/delete.go

```diff
@@ -1,33 +1,46 @@
 package delete
 
+//This is going to declare appropriate struct based on destination type & call `Deleter` method of it.
+//to get deletion done.
+//called by JobSvc with (model.Job, model.Destination).
+//returns final status,error ({successful, failure}, err)
+
 import (
 	"context"
-	"fmt"
 
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
 )
 
-type deleter interface {
-	Delete(ctx context.Context, job model.Job, dest model.Destination) (model.JobStatus, error)
+type apiManager interface {
+	Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus
+}
+
+type batchManager interface {
+	Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus
+}
+
+type customManager interface {
+	Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus
 }
```

Comment:
Those interface are similar, why not  using a single one
```suggestion
type deleter interface {
	Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus
}
```